src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/AuthorisationImpl.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.darts.authorisation.component.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.audio.repository.MediaRequestRepository;
+import uk.gov.hmcts.darts.authorisation.api.AuthorisationApi;
+import uk.gov.hmcts.darts.authorisation.component.Authorisation;
+import uk.gov.hmcts.darts.cases.repository.CaseRepository;
+import uk.gov.hmcts.darts.common.entity.CourthouseEntity;
+import uk.gov.hmcts.darts.common.enums.SecurityRoleEnum;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
+import uk.gov.hmcts.darts.common.repository.HearingRepository;
+import uk.gov.hmcts.darts.common.repository.MediaRepository;
+import uk.gov.hmcts.darts.common.repository.TranscriptionRepository;
+
+import java.util.List;
+import java.util.Set;
+
+import static uk.gov.hmcts.darts.audio.exception.AudioApiError.MEDIA_NOT_FOUND;
+import static uk.gov.hmcts.darts.audio.exception.AudioRequestsApiError.MEDIA_REQUEST_NOT_FOUND;
+import static uk.gov.hmcts.darts.cases.exception.CaseApiError.CASE_NOT_FOUND;
+import static uk.gov.hmcts.darts.hearings.exception.HearingApiError.HEARING_NOT_FOUND;
+import static uk.gov.hmcts.darts.transcriptions.exception.TranscriptionApiError.TRANSCRIPTION_NOT_FOUND;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class AuthorisationImpl implements Authorisation {
```

Comment:
Just an idea, no need to action it. 

This implementation will end up becoming a 'god' type mediator class. We could have an authorisation interface that each service implements? This way we decentralize this implementation, ensuring that authorisation for a specific resource is the responsibility of a specific service class.

Reply:
@hemantasharma1129 advises to keep it for now. It was mainly done, so that I could standup a simple test schema to develop against in our integration tests using `AuthorisationStub`. Not all the endpoints can be protected by the custom annotation currently as they may not be acting on a single context ID parameter e.g. POST transcription and GET cases. So they will have to be done on the data layer [DMP-626](https://tools.hmcts.net/jira/browse/DMP-626).